alts/src/main/java/io/grpc/alts/internal/AltsProtocolNegotiator.java

```diff
@@ -65,24 +64,30 @@
    * channel.
    */
   public static final class ClientAltsProtocolNegotiatorFactory
-      implements InternalNettyChannelBuilder.ProtocolNegotiatorFactory {
+      implements InternalProtocolNegotiator.ClientFactory {
 
     private final ImmutableList<String> targetServiceAccounts;
-    private final LazyChannel lazyHandshakerChannel;
+    private final ObjectPool<Channel> handshakerChannelPool;
 
     public ClientAltsProtocolNegotiatorFactory(
         List<String> targetServiceAccounts,
         ObjectPool<Channel> handshakerChannelPool) {
       this.targetServiceAccounts = ImmutableList.copyOf(targetServiceAccounts);
-      this.lazyHandshakerChannel = new LazyChannel(handshakerChannelPool);
+      this.handshakerChannelPool = handshakerChannelPool;
     }
 
     @Override
-    public ProtocolNegotiator buildProtocolNegotiator() {
+    public ProtocolNegotiator newNegotiator() {
+      LazyChannel lazyHandshakerChannel = new LazyChannel(handshakerChannelPool);
       return new ClientAltsProtocolNegotiator(
```

Comment:
What do you think of moving the creation of `ClientTsiHandshakerFactory` and `LazyChannel` to inside of `ClientAltsProtocolNegotiator` constructor? The reason is `lazyHandshakerChannel` is closed inside ``ClientAltsProtocolNegotiator.close()` so it looks symmetric to create and close (release) `lazyHandshakerChannel` in that class. Then `ClientAltsProtocolNegotiator` also will need to be created there. Just a suggestion.

Reply:
Done.